src/analysis/complex/polynomial.lean

```diff
@@ -103,3 +104,37 @@ lt_irrefl (f.eval z₀).abs $
     ... = (f.eval z₀).abs : sub_add_cancel _ _
 
 end complex
+
+section real
+
+theorem nat_degree_eq_zero_of_derivative_eq_zero {f : polynomial ℝ} (h : f.derivative = 0) :
```

Comment:
Please move this to `data.polynomial.derivative` (or whichever file contains the definition of derivative) and generalize it to arbitrary integral domains of characteristic 0.

Reply:
I have moved it to `degree.lean` as it is about degree. Will this be acceptable?

Reply:
Generalised version of course

Reply:
It should go in `derivative.lean` imo.

See https://leanprover-community.github.io/mathlib_docs/data/polynomial/derivative.html#polynomial.degree_derivative_eq (and 3 more lemmas about `degree`) in `derivative.lean`.

Reply:
I've changed it, thank you for your code.